code/modules/mob/living/living_emote.dm

```diff
@@ -11,6 +11,10 @@
 	. = ..()
 	if(user.mind?.miming)
 		return FALSE  // shh
+	if(user.is_muzzled())
```

Comment:
Careful -- this will mute all living (and descendent) emotes if someone is muzzled. What I think you want is to extract it to an override that only applies to this deathgasp type.

You probably want something like:

```
/datum/emote/living/deathgasp/should_play_sound(mob/user, intentional)
    . = ..()
    if(user.is_muzzled() && intentional)
         return FALSE
```

Reply:
Sorry for taking so long! Exams and IRL shenanigans kept me busy. 

Applied this method and it works perfectly.

Many thanks for the help Lewcc! ^^